app/models/piece.rb

```diff
@@ -3,6 +3,20 @@ class Piece < ActiveRecord::Base
 
   enum color: [:white, :black]
 
+  def move_to!(new_x, new_y)
+    piece = Game.find(game_id).pieces.find_by(x_coordinate: new_x, y_coordinate: new_y)
```

Comment:
Also, instead of naming this variable `piece`. I'd recommend something more descriptive.

How about, `destination_piece` ?